camayoc/tests/qcs/cli/test_scans.py

```diff
@@ -0,0 +1,332 @@
+        'id': scan_id,
+        'result': None,
+    })
+    assert 'connection_results' in result
+    assert 'inspection_results' in result
+
+
+def test_scan_with_disabled_products(isolated_filesystem, qpc_server_config):
+    """Perform a scan and disable an optional product.
+
+    :id: b1cd9901-44eb-4e71-846c-34e1b19751d0
+    :description: Perform a scan and disable an optional product.
+    :steps: Run ``qpc scan start --sources <source> --disable-optional-products
+        <optional-product>``
+    :expectedresults: The scan must completed without any error and a report
+        should be available. The disabled products should not have results in
+        the report.
+    :caseautomation: notautomated
+    """
+
+
+def test_scan_restart(isolated_filesystem, qpc_server_config):
+    """Perform a scan and ensure it can be paused and restarted.
+
+    :id: 7eb79aa8-fe3d-4fcd-9f1a-5e2d4df2f3b6
+    :description: Start a scan, then pause it and finally restart it.
+    :steps:
+        1) Run ``qpc scan start --sources <source>`` and store its ID.
+        2) Stop the scan by running ``qpc scan stop --id <id>``
+        3) Restart the scan by running ``qpc scan restart --id <id>``
+    :expectedresults: The scan must completed without any error and a report
+        should be available.
+    """
+    result = scan_start({
+        'source': config_sources()[0]['name'],
+    })
+    match = re.match(r'Scan "(\d+)" started.', result)
+    assert match is not None
+    scan_id = match.group(1)
+    wait_for_scan(scan_id, status='running')
+    scan_pause({'id': scan_id})
+    wait_for_scan(scan_id, status='paused')
+    scan_restart({'id': scan_id})
+    wait_for_scan(scan_id)
+    result = scan_show({
+        'id': scan_id,
+        'result': None,
+    })
+    assert 'connection_results' in result
+    assert 'inspection_results' in result
+
+
+def test_scan_cancel(isolated_filesystem, qpc_server_config):
+    """Perform a scan and ensure it can be canceled.
+
+    :id: b5c11b82-e86e-478b-b885-89a577f81b13
+    :description: Start a scan, then cancel it and finally check it can't be
+        restarted.
+    :steps:
+        1) Run ``qpc scan start --sources <source>`` and store its ID.
+        2) Cancel the scan by running ``qpc scan cancel --id <id>``
+        3) Try to restart the scan by running ``qpc scan restart --id <id>``
+    :expectedresults: The scan must be canceled and can't not be restarted.
+    """
+    result = scan_start({
+        'source': config_sources()[0]['name'],
+    })
+    match = re.match(r'Scan "(\d+)" started.', result)
+    assert match is not None
+    scan_id = match.group(1)
+    wait_for_scan(scan_id, status='running')
+    scan_cancel({'id': scan_id})
+    wait_for_scan(scan_id, status='canceled')
+    result = scan_restart({'id': scan_id}, exitstatus=1)
+    assert result.startswith(
+        'Error: Scan cannot be restarted. The scan must be paused for it to '
+        'be restarted.'
+    )
```

Comment:
May want to add a test case for cancelling a paused scan.

Reply:
Good one, will add